src/Lib/Db.hs

```diff
@@ -1,54 +1,53 @@
-{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ConstraintKinds #-}
 
 module Lib.Db
-       ( queryPG
-       , queryPG_
-       , executePG
-       , executePG_
-       , executeManyPG
+       ( query
+       , query_
+       , execute
+       , execute_
+       , executeMany
+       , asSingleRow
        ) where
 
 import Lib.App.Env (AppEnv (..))
+import Lib.App.Error (WithError, notFoundOnNothingM)
 
 import qualified Data.Pool as Pool
-import qualified Database.PostgreSQL.Simple as PG
+import qualified Database.PostgreSQL.Simple as SQL
+
+type WithDbPool m = (MonadReader AppEnv m, MonadIO m)
 
 -- Query the database with a given query and args and expect a list of
 -- rows in return
-queryPG :: (MonadReader AppEnv m, MonadIO m, PG.ToRow q, PG.FromRow r) =>
-  PG.Query -> q -> m [r]
-queryPG query args = do
-  pool <- asks dbPool
-  liftIO $ Pool.withResource pool (\conn -> PG.query conn query args)
+query :: (WithDbPool m, SQL.ToRow q, SQL.FromRow r) => SQL.Query -> q -> m [r]
+query qx args = perform (\conn -> SQL.query conn qx args)
 
 -- Query the database with a given query and no args and expect a list
 -- of rows in return
-queryPG_ :: (MonadReader AppEnv m, MonadIO m, PG.FromRow r) =>
-  PG.Query -> m [r]
-queryPG_ query = do
-  pool <- asks dbPool
-  liftIO $ Pool.withResource pool (`PG.query_` query)
+query_ :: (WithDbPool m, SQL.FromRow r) => SQL.Query -> m [r]
+query_ qx = perform (`SQL.query_` qx)
 
 -- Query the database with the given query and args, returning the
 -- number of rows affected
-executePG :: (MonadReader AppEnv m, MonadIO m, PG.ToRow q) =>
-  PG.Query -> q -> m Int64
-executePG query args = do
-  pool <- asks dbPool
-  liftIO $ Pool.withResource pool (\conn -> PG.execute conn query args)
+execute :: (WithDbPool m, SQL.ToRow q) => SQL.Query -> q -> m Int64
+execute q args = perform (\conn -> SQL.execute conn q args)
 
 -- Query the database with the given query and no args, returning the
 -- number of rows affected
-executePG_ :: (MonadReader AppEnv m, MonadIO m) =>
-  PG.Query -> m Int64
-executePG_ query = do
-  pool <- asks dbPool
-  liftIO $ Pool.withResource pool (`PG.execute_` query)
+execute_ :: (WithDbPool m) => SQL.Query -> m Int64
+execute_ qx = perform (`SQL.execute_` qx)
 
 -- Query the database many times with a given query and a list of
 -- args. Returns the number of rows affected
-executeManyPG :: (MonadReader AppEnv m, MonadIO m, PG.ToRow q) =>
-  PG.Query -> [q] -> m Int64
-executeManyPG query args = do
-  pool <- asks dbPool
-  liftIO $ Pool.withResource pool (\conn -> PG.executeMany conn query args)
+executeMany :: (WithDbPool m, SQL.ToRow q) => SQL.Query -> [q] -> m Int64
+executeMany qx args = perform (\conn -> SQL.executeMany conn qx args)
+
+-- Helper function working with results from a database when you expect
```

Comment:
Could you add `|` after `--` in each comment before function?

Comments like this:

```haskell
-- | Some comment
```

are Haddock comments. They will be shown in generated documentation unlike any other comments. So it's usually a good idea to write Haddock comments before top-level function. If you see function that has comment but it's not `-- |` comment, you can fix this as well 🙂